en/flight_modes/return.md

```diff
@@ -2,33 +2,107 @@
 
-A copter/multi-rotor will first ascend to the return altitude (*by default*, [RTL_RETURN_ALT](#RTL_RETURN_ALT)) and then fly to the home position in a straight line and constant altitude (if already above the return altitude it will return at its current altitude).
+If a mission landing has been defined, the vehicle will:
+- Ascend to a safe [return altitude](#return_altitude) above any expected obstacles.
+- Fly a direct path to the start of a mission landing sequence
+- Descend/ascend to the specified waypoint altitude
+- Execute the landing sequence.
 
-When it arrives at the home/launch position it will rapidly descend to the [RTL_DESCEND_ALT](#RTL_DESCEND_ALT) altitude.
-It will then hover for the period defined in [RTL_LAND_DELAY](#RTL_LAND_DELAY) before landing.
+If no mission landing is defined PX4 will revert to a [Safety Point Return](#safety_point_return).
 
-The required return altitude can be configured using using the parameters [RTL_RETURN_ALT](#RTL_RETURN_ALT) and [RTL_CONE_ANG](#RTL_CONE_ANG) (which define a half cone centered around the home position).
+
+### Mission Path Return Type (RTL_TYPE=2) {#mission_path_return}
+
+This return type uses the mission (if defined) to provide a safe return *path*, and the mission landing pattern (if defined) to provide landing behaviour.
+If there is a mission but no mission landing pattern, the mission is flown *in reverse*.
+
+> **Note** The behaviour is fairly complex because it depends on the flight mode, and whether a mission and mission landing are defined.
+
+Mission *with* landing pattern:
+- **Mission mode:** Mission is continued in "fast-forward mode" (jumps, delay and any other non-position commands ignored, loiter and other position waypoints converted to simple waypoints) and then lands.
+- **Auto mode other than mission mode:**
+  - Ascend to a safe [return altitude](#return_altitude) above any expected obstacles.
+  - Fly directly to closest waypoint (for FW not a landing WP) and descend to waypoint altitude.
+  - Continue mission in fast forward mode from that waypoint.
+- **Manual modes:** Same as [Mission Landing Return](#mission_landing_return).
+  - Ascend to a safe [return altitude](#return_altitude) above any expected obstacles.
+  - Fly directly to landing sequence position and descend to waypoint altitude
+  - Land using mission landing pattern
+
+Mission *without* landing pattern defined:
+- **Mission mode:** 
+  - Mission flown "fast-backward" (in reverse) starting from the previous waypoint 
+    - Jumps, delay and any other non-position commands ignored, loiter and other position waypoints converted to simple waypoints.
+    - VTOL vehicles transition to FW mode (if needed) before flying the mission in reverse.
+  - On reaching waypoint 1, the vehicle ascends to the [return altitude](#return_altitude) and flies to the home position (where it [lands or waits](#arrival)).
```

Comment:
Can this be confirmed. Ie if I get to waypoint 1, it then ascends to RTL altitude and flies home - ie it effectively treats the first waypoint as the beginning of a normal RTL.

What if waypoint=1 is a takeoff waypoint? Will it reverse that and land?

Reply:
@hamishwillee I think you are right, but let me double check the takeoff waypoint.

Reply:
I tried this and takeoff waypoint just gets treated like any other waypoint. It heads to it, and when it gets acceptance it then goes back to home. This therefore is strictly correct.